### ISSUE TITLE
metrics with 0 value ( also false ) are mapped to NaN

### ISSUE THREAD

**Issue by abenbouz** [marcusolsson/grafana-json-datasource#25]
hello Marc ,

can you please help resolving the floowing issue :
metrics with 0 value ( also false ) are mapped to NaN
ENV : 
- Grafana version 7.3.4 
- grafana-json-datasource version 0.7

FYI : mapping is working well with  previous version ( 0.6.4 )

Regards

**Comment by marcusolsson** [comment]
Thank you for reporting this! I'll look into it 👍

**Comment by abenbouz** [comment]
thanks for you prompt action 
i forget to mention that i'm using Table plugin

**Comment by marcusolsson** [comment]
Could you paste the JSON and the query you are using, so that I reproduce this?

**Comment by abenbouz** [comment]
i'm using a private backend so you don't have acess to it 
but will give you some outputs 👍 

Path : /client  Query string : type=X&attributes=*

$.client[?(@.name=='$name')].name
$.client[?(@.name=='$name')].status.child

{
  "request": {
    "url": "api/datasources/proxy/9/client?type=X&attributes=*",
    "method": "GET",
    "hideFromInspector": false
  },
  "response": {
    "client": [
      {
        "name": "XXXXX",
        "status": {
          "child": 0
        },
        "type": "X"
      } 
	 ]
    }
}

anyway you can reproduce it using any data with 0 as a value

**Comment by marcusolsson** [comment]
I tried with this JSON:

```
[
  {"number": 0, "boolean": false},
  {"number": 2, "boolean": true}
]
```

which seems to work

<img width="553" alt="Screen Shot 2020-12-07 at 16 26 20" src="https://user-images.githubusercontent.com/8396880/101369748-f326ce80-38a8-11eb-9aab-10d2badc7bea.png">

**Comment by marcusolsson** [comment]
I can reproduce it with your example.

**Comment by abenbouz** [comment]
i don't think it related to the number of items

**Comment by abenbouz** [comment]
since i have the same behaviour with a lot of items

**Comment by marcusolsson** [comment]
Yeah, you're right. 

**Zeros:**

- `[0, 0, 0]` becomes `[NaN, NaN, NaN]`
- `[0, 0, 1]` becomes `[0, 0, 1]`

**Booleans:**

- `[false, false, false]` becomes `[NaN, NaN, NaN]`
- `[false, false, true]` becomes `[false, false, true]`

Looks like it fails when all values are "falsy".

**Comment by abenbouz** [comment]
in fact in depends on the selct Json Path 
if you select just one item based on some creteria you have the problem

**Comment by abenbouz** [comment]
and if you do not  choose any filter like (?(@.name=='$name') it should be ok

**Comment by abenbouz** [comment]
and please not that it was working fine in previous version

**Comment by marcusolsson** [comment]
I've found the issue, and working on a fix.

**Comment by abenbouz** [comment]
perfect thanks

**Comment by marcusolsson** [comment]
Fixed in [v0.7.1](https://grafana.com/grafana/plugins/marcusolsson-json-datasource?version=0.7.1). Let me know if you're still having issues!

**Comment by abenbouz** [comment]
i apreciate well your reactivity :)
V0.7.1 is working like a charm 
Thanks
